src/modes/coincident_select.js

```diff
@@ -83,13 +83,26 @@ CoincidentSelect.onSetup = function(opts) {
   const planId = features.find(f => f.properties.vetro_id === feature.id)
     .properties.plan_id;
 
-  features.forEach(f => {
+  for(const f of features){
     if (
       f.properties.plan_id === planId &&
       f.geometry.type === "LineString" &&
       !f.layer.id.includes("_snap")
     ) {
-      const lineGeom = JSON.parse(f.properties.geojson_string);
+
+      let lineGeom = f.geometry;
+      if(this._ctx.options.fetchSourceGeometry && typeof this._ctx.options.fetchSourceGeometry === "function"){
+        const lineSrcGeom = await this._ctx.options.fetchSourceGeometry(f);
+        if(lineSrcGeom && lineSrcGeom.type && lineSrcGeom.coordinates.length){
+          lineGeom = lineSrcGeom;
+        }
+
+        const ptSrcGeom = await this._ctx.options.fetchSourceGeometry({ properties: { vetro_id: state.initiallySelectedFeatureIds[0] }});
```

Comment:
Can this and the above this._ctx.options.fetchSourceGeometry call be run in parallel?